products/cloudiot/api.yaml

```diff
@@ -192,3 +192,213 @@ objects:
                       output: true
                       description: |
                         The type of public key in the certificate.
+  - !ruby/object:Api::Resource
+    name: 'Device'
+    base_url: '{{registry}}/devices'
+    self_link: '{{registry}}/devices/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Google Cloud IoT Core device.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/iot/docs/'
+      api: 'https://cloud.google.com/iot/docs/reference/cloudiot/rest/'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: registry
+        input: true
+        url_param_only: true
+        required: true
+        description: |
+          The name of the device registry where this device should be created.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'id'
+        input: true
+        required: true
+        description: |
+          The unique identifier for the device. For example,
+          `Device0`.
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource path name. For example,
+          `projects/example-proj/locations/us-central1/registries/my-registry/devices/device0`.
+      - !ruby/object:Api::Type::String
+        name: 'numId'
+        output: true
+        description: |
+          A server-defined unique numeric ID for the device.
+          This is a more compact way to identify devices, and it is globally unique.
+      - !ruby/object:Api::Type::Array
+        name: 'credentials'
+        description: |
+          The credentials used to authenticate this device.
+        max_size: 3
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::Time
+              name: 'expirationTime'
+              allow_empty_object: true
+              output: true
+              description: |
+                The time at which this credential becomes invalid.
+            - !ruby/object:Api::Type::NestedObject
+              name: 'publicKey'
+              description: |
+                A public key used to verify the signature of JSON Web Tokens (JWTs).
+              properties:
+                - !ruby/object:Api::Type::Enum
+                  name: 'format'
```

Comment:
Should this field and the `key` field be required?

Reply:
done